File: scripts/refresh_ingredients_taxonomy.mjs
import { writeFileSync } from 'fs';
import stringify from 'json-stable-stringify';

fetch('https://static.openfoodfacts.org/data/taxonomies/ingredients.json').then(async (response) => {
    writeFileSync('ciqual/ingredients.json', stringify(await response.json(), {space: 2}));
});

File: scripts/refresh_ingredients_taxonomy.py
import os
import urllib.request, json 
with urllib.request.urlopen("https://static.openfoodfacts.org/data/taxonomies/ingredients.json") as url:
    data = json.load(url)
    filename = os.path.join(os.path.dirname(__file__), "../recipe_estimator/assets/ingredients.json")
    with open(filename, 'w', encoding='utf-8') as f:
      f.write(json.dumps(data, ensure_ascii=False, sort_keys=True, indent=2))
